### PR TITLE
Enable parallel adapter inference together with attention_mask

### PR DESCRIPTION
**Pull request by calpt**
Follow-up to #116:

In the current parallel inference implementation, it is not possible to set an attention mask at the same time, making parallel inference difficult to use in practice. This PR adds a new hook (`adjust_attention_mask_for_parallel()`) called after every Transformers layer that ensures that the attention mask is properly resized when the hidden states are.

This is enough to make it work for BERT, RoBERTa & DistilBERT, but the Parallel block might not be usable for other model types (e.g. BART). Therefore, I added a whitelist for this block in `adapter_composition.py`.

### CODE REVIEW

File: tests/test_adapter_composition.py [hunk 1]
```diff
@@ -118,6 +118,7 @@ def test_parallel_inference_with_heads(self):
 
             with self.subTest(model_class=model_class.__name__):
                 inputs = {}
+                inputs["attention_mask"] = torch.ones((2, 128))
```
Comment (arueckle) [inline tests/test_adapter_composition.py:121]: can we have a different attention mask for the different inputs? this would verify whether the attention masks are repeated correctly 

With the current test case we could not detect a difference between the two types of repetitions because a==b
(a,b) -> (a,a,b,b)
(a,b) -> (a,b,a,b)

**Review by arueckle — CHANGES_REQUESTED** [review]
Overall, this looks much cleaner compared to the previous version. 
I think the testcase needs to be adapted slightly to also test for the correctness of the repetition of the attention mask.

File: tests/test_adapter_composition.py (hunk 1 above)
Comment (calpt) [reply tests/test_adapter_composition.py]: Updated to use a random attention mask for each input.

**Review by arueckle — APPROVED** [review]
